sonar-db/src/main/java/org/sonar/db/component/ComponentMapper.java

```diff
@@ -141,7 +141,12 @@
 
   void insertBatch(ComponentDto componentDto);
 
-  void update(ComponentDto componentDto);
+  void update(ComponentUpdateDto component);
+
+  void applyBChangesForRootComponentUuid(@Param("projectUuid") String projectUuid);
+
+  void resetBChangedForRootComponentUuid(@Param("projectUuid") String projectUuid);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document this public method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUndocumentedApi)